core-java-modules/core-java-streams-5/src/test/java/com/baeldung/aggregateEx/AggregateExceptionHandlerUnitTest.java

```diff
@@ -0,0 +1,180 @@
+            return new RuntimeException(e);
+        }
+    }
+
+    @Test
+    public void givenExtractedMethod_whenFoundNonInt_thenAggregateException() {
+        String[] strings = {"1", "2", "3", "a", "b", "c"};
+        RuntimeException runEx = Arrays.stream(strings)
+                .map(AggregateExceptionHandlerUnitTest::process)
+                .filter(Objects::nonNull)
+                .reduce(new RuntimeException("Errors Occurred"), (o1, o2) -> {
+                    o1.addSuppressed(o2);
+                    return o1;
+                });
+        assertEquals("Errors Occurred", runEx.getMessage());
+        assertEquals(3, runEx.getSuppressed().length);
+    }
+
+    @Test
+    public void givenTryCatchInPipeline_whenFoundNonInts_thenAggregateException() {
+        String[] strings = {"1", "2", "3", "a", "b", "c"};
+        RuntimeException runEx = Arrays.stream(strings)
+                .map(str -> {
+                    try {
+                        Integer.parseInt(str);
+                        return null;
+                    } catch (NumberFormatException e) {
+                        return new RuntimeException(e);
+                    }
+                })
+                .filter(Objects::nonNull)
+                .reduce(new RuntimeException("Errors Occurred"), (o1, o2) -> {
+                    o1.addSuppressed(o2);
+                    return o1;
+                });
+        assertEquals("Errors Occurred", runEx.getMessage());
+        assertEquals(3, runEx.getSuppressed().length);
+    }
+
+    @Test
+    public void whenFoundNonInts_thenAggregateExceptionAndReturnOutput() {
+        String[] strings = {"1", "2", "3", "a", "b", "c"};
+        Map resultMap = Arrays.stream(strings)
+                .map(AggregateExceptionHandlerUnitTest::transform)
+                .collect(Collectors.partitioningBy(o -> o instanceof RuntimeException, Collectors.toList()));
+        RuntimeException ex = null;
+        if (resultMap.containsKey(Boolean.TRUE)) {
+            List<RuntimeException> exs = (List<RuntimeException>) resultMap.get(Boolean.TRUE);
+            ex = exs.stream()
+                    .reduce(
+                            new RuntimeException("Errors Occurred"), (o1, o2) -> {
+                                o1.addSuppressed(o2);
+                                return o1;
+                            });
+        }
+        assertEquals("Errors Occurred", ex.getMessage());
+        assertEquals(3, ex.getSuppressed().length);
+    }
+
+    @Test
+    public void givenWrapFunction_whenFoundNonInts_thenAggregateException() throws ExceptionAggregator {
+        String[] strings = {"1", "2", "3", "a", "b", "c"};
+        Map<Boolean, List<Result<Integer>>> resultmap = Arrays.stream(strings)
+                .map(CustomMapper.mapper(Integer::parseInt))
+                .collect(Collectors.partitioningBy(r -> r.getException().isEmpty(), Collectors.toList()));
+
+        if (resultmap.containsKey(Boolean.FALSE)) {
```

Comment:
Replace the if statement with an assertion instead

Reply:
agreed